tests/data/test_dataloader.py

```diff
@@ -428,27 +428,99 @@ def test_tokenize_example_malformed():
         'response': 'response',
```

Comment:
Can you also add a dataloader test that loads chat formatted data? Similar to https://github.com/mosaicml/llm-foundry/blob/main/tests/data/test_dataloader.py#L255